### DISCUSSION TITLE
Finder Tabs does not work

### DISCUSSION THREAD

**Discussion by zisoft** [ttscoff/bunch-docs#88] [General]
bunch is great, thank you for that!

I want to open Finder with multiple tabs. According to the documentation I tried to use the -> syntax, but this does not work.
This is my simle bunch file to test:

```
% Finder
- XX
- ~/Desktop
-> ~/Documents

```
So this should first close all Finder windows, open Finder with the `~/Desktop` folder and open a new tab with the `~/Documents` folder.

But this does nothing, no Finder window comes up.

When I change

`-> ~/Documents`
to
`- ~/Documents`

it opens Finder with two separate windows which is correct.

Is this a bug or am I doing something wrong?

**Comment by ttscoff** [comment]
Your example works properly for me, but Finder tabs basically just uses 
AppleScript to create tabs with Accessibility scripting, so there's a 
lot that could go wrong.

The most likely culprit is a lack of accessibility permissions. Bunch 
should have requested them when it needed them, but just to be sure 
they're available, open System Preferences -> Security and Privacy -> 
Accessibility and ensure that Bunch.app is in the list and has a 
checkmark. Let me know if that helps.

-Brett

**Reply by zisoft** [reply]
Unfortunately this does not help. Bunch is listed in the Accessibility preferences and it does indeed ask for permission when trying to access a folder for the first time.

When this is done via AppleScript, does AppleScript need to be listed in the security preferences as well?

**Reply by zisoft** [reply]
ah, I found the following in the logs:

```
2021-06-29 22:21:08: 🔴[test] {
    NSAppleScriptErrorAppName = "System Events";
    NSAppleScriptErrorBriefMessage = "Can\U2019t get menu bar item \"File\" of menu bar 1 of application process \"Finder\".";
    NSAppleScriptErrorMessage = "System Events got an error: Can\U2019t get menu bar item \"File\" of menu bar 1 of application process \"Finder\".";
    NSAppleScriptErrorNumber = "-1728";
    NSAppleScriptErrorRange = "NSRange: {0, 0}";
}

```
I'm running the german locale, the menu bar item "File" is called "Ablage" in german.

**Comment by ttscoff** [comment]
Ah, yeah, that's going to get more complicated than I'll have time to 
look into until the weekend. I could probably safely refer to that menu 
by its index in the menu bar, but I'll need to do some testing.

-Brett

**Reply by Younghart** [reply]
So it's a localization issue, too! I was already wondering since I also can't get Finder to open a location in tabs (no matter how long I wait). And, as you may have already guessed, I'm also using localized macOS (German).
Localized strings seem to be a mess in AppleScript.

**Reply by hoeni** [reply]
Is there a solution or workaround for this already? Just run into the same issue (MacOS in German).

**Reply by hoeni** [reply]
Even a 'hackish' solution like a preference setting for the Finder menu string (defaulting to the English version) would make things useable in this case...

**Reply by Younghart** [reply]
I cobbled together a small workaround. It starts with creating my own keyboard shortcut in System Preferences/Keyboards/Shortcuts. Since my Mac is also set to German, I created a Finder-specific keyboard shortcut for "Alle Fenster zusammenführen" (see picture). The spelling has to be exactly as in the menu. To confirm its function, I switched back to Finder after saving the shortcut, opened the "Fenster" menu – if it works, the shortcut appears next to "Alle Fenster zusammenführen".

![Finder_Merge_Windows_Keyboard_Shortcut](https://user-images.githubusercontent.com/2934471/190001004-62314b9c-7f17-480b-857a-ede3bf666c05.png)

This keyboard shortcut I then added to my Bunch file after all the paths I wanted to add , in curly brackets:

```
Finder
- XX
- ~/Desktop
- ~/Documents/Project01
- ~/Library/Mobile Documents/com~apple~CloudDocs/ABC
- {^~@m}
```

This way, I can only create one tabbed group of Finder windows, but that's enough for my purposes. 

I hope this helps!

**Reply by hoeni** [reply]
That's a great idea! However, in my case, I indeed needed to open many Finder windows with many tabs each for sorting files to many destinations...
